pages/app/settings.vue

```diff
@@ -162,20 +169,28 @@ const onSelectProfilePhoto = (event) => {
     event.target.value = '' // Reset the input
     return
   }
-
+  imagePreview.value = URL.createObjectURL(file)
   storeUser.setProfilePic(file)
 }
 
 const onClickUploadPhoto = () => {
   profilePhotoInput.value.click()
 }
 
+const onClickClearPhoto = async () => {
+  await storeUser.clearProfilePic()
+  imagePreview.value = null
+}
+
 onMounted(() => {
-  let unwatch = null
-  unwatch = watch(() => storeUser.isReady, (value) => {
+  const unwatch = watch(() => storeUser.isReady, (value) => {
     if (value) {
       syncFormData()
-      unwatch && unwatch()
+      storeUser.fetchProfilePic(storeUser.getProfile.id)
+        .then(base64Image => {
+          imagePreview.value = base64Image
+          unwatch()
```

Comment:
You shouldn't use unwatch within then statement here. Because, if fetch profile gets failed, this always will be watching. If fetch profile pic fails, what are the chances of it is being sucess?

So, you should write unwatch outside the promise itself.